ls1intum/Artemis
# ImproveArtemisDurationFromSecondsPipe to support more time formats

### Checklist

### Motivation and Context


Fix #3692

### Description

This PR changes the ArtemisDurationFromSecondsPipe to handle time spans that are multiple hours or even days long. The days and hours are left out, if they would only show 0.

This PR additionally fixes a type error that occurred in the refactored code.

### Steps for Testing


1. Log in to Artemis
2. Create an exam that has a long working time (>= 60 Minutes)
3. Participate as a student and check all occurrences of a duration for their correctness.
- The exam card in the exams tab of a course
- The exam cover sheet (If the student has additional working time, the amount of time added should be displayed in brackets)
- The timer in the top right of the exam and the hand in screen (The timer should only show the seconds when it is below ten minutes to not distract the students too much during the exam)
4. Check the localization of "day"/"Tag", if the exam spans multiple days

For the type error:
1. Go to the student exams and select one
2. View the submission of the exam
3. Check the console and make sure that no type error occurs.

### Test Coverage




artemis-duration-from-seconds.pipe.ts: Now tested 100%

### Screenshots


Exam timer:
![Bildschirmfoto 2021-07-28 um 18 21 33](https://user-images.githubusercontent.com/38322605/127360770-358d7ff9-c65d-443d-b154-249c6c12d945.png)
![Bildschirmfoto 2021-07-28 um 18 22 02](https://user-images.githubusercontent.com/38322605/127360783-fbb4d0ba-a52c-4a66-a071-a462e316c9b7.png)

Exam overview:
![Bildschirmfoto 2021-07-29 um 12 53 14](https://user-images.githubusercontent.com/38322605/127479892-554b095e-d0c1-4e7a-a4b5-97a92d808cd4.png)

Cover page (with additional working time displayed in brackets):
![Bildschirmfoto 2021-07-29 um 12 52 47](https://user-images.githubusercontent.com/38322605/127479884-8597ed2f-a493-4362-9ac0-26d580dda98f.png)
